packages/core/src/util/txInspector.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  Address,
+  CertificateType,
+  Ed25519KeyHash,
+  Lovelace,
+  PoolId,
+  StakeAddressCertificate,
+  StakeDelegationCertificate,
+  TokenMap,
+  TxAlonzo,
+  TxOut,
+  Value
+} from '../Cardano';
+
+const isMyAddress =
+  (addresses: Address[]) =>
+  ({ address }: { address: Address }) =>
+    addresses.includes(address!);
+
+const isOutgoing = (tx: TxAlonzo, ownAddresses: Address[]): boolean => tx.body.inputs.some(isMyAddress(ownAddresses));
+
+const getTotalValue = (outputs: TxOut[]): Value => {
+  const result: Value = { coins: 0n };
+  const totalAssets: TokenMap = new Map();
+
+  for (const output of outputs) {
+    result.coins += output.value.coins;
+    if (output.value.assets) {
+      for (const [asset, amount] of output.value.assets) {
+        const total = totalAssets.get(asset) ?? 0n;
+        totalAssets.set(asset, total + amount);
+      }
+    }
+  }
+
+  if (totalAssets.size > 0) result.assets = totalAssets;
+  return result;
+};
+
+type InspectionResult = { [k: string]: unknown } | null;
+type InspectTx = (tx: TxAlonzo) => InspectionResult;
+type Inspectors = { [k: string]: InspectTx };
+type TxInspector<T extends Inspectors> = (tx: TxAlonzo) => {
+  [k in keyof T]: ReturnType<T[k]>;
+};
+
+export type SendReceiveCoinInspector = {
```

Comment:
It's not just coin, but also assets. Also think the 'inspector' is a function, and this is a result?
I think returning `{coins: 0n}` is nice than null for this inspector.
```suggestion
export type SendReceiveValue = Cardano.Value;
```

Reply:
I vote for just typing the return value from the inspector as `Cardano.Value`